dxr/plugins/xbl/analyzer.py

```diff
@@ -0,0 +1,91 @@
+from warnings import warn
+import xml.parsers.expat as expat
+
+from dxr.indexers import Extent, Position
+from dxr.plugins.xbl.refs import PLUGIN_NAME, TypeRef
+
+
+class XBLAnalyzer(object):
+    def __init__(self, path, tree, contents, encoding):
+        self.tree = tree
+        self.path = path
+        self.contents = contents
+        self.lines = contents.splitlines(True)
+        self.parser = expat.ParserCreate(encoding)
+        # Stack of namespaces, issued by the 'id' attr of bindings.
+        self.namespace = []
+        self.needles = []
+        self.refs = []
+        # Late bind our custom handlers to the parser
+        self.parser.StartElementHandler = self.StartElementHandler
+        self.parser.EndElementHandler = self.EndElementHandler
+        try:
+            self.parser.Parse(self.contents.encode(encoding))
+        except expat.ExpatError as err:
+            warn('Exception occurred on parsing XBL file {}'.format(self.path.encode('utf-8')))
+            warn(err.message)
+
+    def extent_for_name(self, name):
+        """Return Extent for the next occurrence of name."""
+        # Because the parser does not provide location info on attr positions,
+        # we can only search for them from the current position.
+        # First check the current line.
+        offset = self.lines[self.parser.CurrentLineNumber - 1].find(name, self.parser.CurrentColumnNumber - 1)
```

Comment:
If the start of the line, would it be very hard to scan backward (rfind) until we find the first ">"? (I'm not sure if this would work with CDATA sections, but it's closer.)